modules/store.js

```diff
@@ -12,68 +13,77 @@ const initialState = {
   loadState: {},
 };
 
-export const reducer = handleActions({
-
-  [BEGIN_GLOBAL_LOAD]: (state) => ({
-    ...state,
-    loaded: false,
-  }),
-
-  [END_GLOBAL_LOAD]: (state) => ({
-    ...state,
-    loaded: true,
-  }),
-
-  [LOAD]: (state, { payload }) => ({
-    ...state,
-    loadState: {
-      ...state.loadState,
-      [payload.key]: {
-        loading: true,
-        loaded: false,
+export const reducer = function reducer(immutableState, action) {
```

Comment:
Probably should just do a wrapper around this reducer instead of how it currently is.